ext/zlib/zlib.c

```diff
@@ -932,10 +948,26 @@ PHP_FUNCTION(inflate_add)
 			break;
 
 		default:
-			php_error_docref(NULL, E_WARNING,
-				"flush mode must be ZLIB_NO_FLUSH, ZLIB_PARTIAL_FLUSH, ZLIB_SYNC_FLUSH, ZLIB_FULL_FLUSH, ZLIB_BLOCK or ZLIB_FINISH");
+			php_error_docref(NULL, E_WARNING, "flush mode must be ZLIB_NO_FLUSH, ZLIB_PARTIAL_FLUSH, ZLIB_SYNC_FLUSH, ZLIB_FULL_FLUSH, ZLIB_BLOCK or ZLIB_FINISH");
 			RETURN_FALSE;
 	}
+
+	if (sizeDelta) {
+		zend_long maxSizeAdd = zval_get_long(sizeDelta);
+
+		if (maxSizeAdd < 0) {
+			php_error_docref(NULL, E_WARNING, "maximum inflated size to increase must be positive, currently " ZEND_LONG_FMT, maxSizeAdd);
+			RETURN_FALSE;
+		}
+
+		php_ctx->maxSize += maxSizeAdd;
+		if (maxSizeAdd < php_ctx->maxSize) { /* overflow check */
+			php_ctx->maxSize = ~0;
+		}
```

Comment:
@bwoebi This overflow check doesn't look right, since if no overflow occurs, the check is always true (n < n+m ∀ m >= 0).